cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/template/etl/api/batch/BatchEndPointStage.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.api.batch;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.template.etl.api.EndPointStage;
+import co.cask.cdap.template.etl.api.PipelineConfigurer;
+
+/**
+ * EndPointStage for Batch Source/Sink.
+ *
+ * @param <T> batch execution context
+ */
+@Beta
+public abstract class BatchEndPointStage<T extends BatchContext> implements EndPointStage {
```

Comment:
It is to refer to the stages at the beginning or the end - ie, source or sink. Any suggestions for better name?